app/react/Viewer/components/OCRDisplay.tsx

```diff
@@ -0,0 +1,190 @@
+            // @ts-ignore
+            socket.on('ocr:error', listenOnError);
+          }
+        })
+        .catch(() => {
+          setOcrStatus({ status: 'ocrError', lastUpdated: Date.now() });
+        });
+    }
+    return () => {
+      // @ts-ignore
+      socket.off('ocr:ready', listenOnSuccess);
+      // @ts-ignore
+      socket.off('ocr:error', listenOnError);
+    };
+  }, []);
+
+  const handleClick = () => {
+    setOcrStatus({ status: 'inQueue', lastUpdated: Date.now() });
+    postToOcr(file.filename || '')
+      .then(() => {
+        // @ts-ignore
+        socket.on('ocr:ready', listenOnSuccess);
+        // @ts-ignore
+        socket.on('ocr:error', listenOnError);
+      })
+      .catch(() => {});
+  };
+
+  const lastUpdated = formatDate(ocrStatus.lastUpdated, locale);
+
+  let statusDisplay = <div />;
+  let tip;
+
+  switch (ocrStatus.status) {
+    case 'loading':
+      statusDisplay = (
+        <div className="status">
+          <p>
+            <Translate>Loading</Translate>&nbsp;...
+          </p>
+        </div>
+      );
+      break;
+
+    case 'noOCR':
+      statusDisplay = (
+        <button type="button" className="btn btn-default" onClick={() => handleClick()}>
+          <Translate>OCR PDF</Translate>
+        </button>
+      );
+      tip = ocrDisplayTips.noOcr();
+      break;
+
+    case 'inQueue':
+      statusDisplay = (
+        <div className="status">
+          <p>
+            <Translate>In OCR queue</Translate>
+          </p>
+        </div>
+      );
+      tip = ocrDisplayTips.lastUpdated(lastUpdated);
+      break;
+
+    case 'unsupported_language':
+      statusDisplay = (
+        <div className="status">
+          <p>
+            <Translate>Unsupported OCR language</Translate>
+          </p>
+        </div>
+      );
+      tip = ocrDisplayTips.unsupportedLang(file.language || 'other');
+      break;
+
+    case 'cannotProcess':
+      statusDisplay = (
+        <div className="status">
+          <p>
+            <Translate>OCR error</Translate>
+          </p>
+        </div>
+      );
+      tip = ocrDisplayTips.cantProcess(lastUpdated);
+      break;
+
+    case 'ocrError':
+      statusDisplay = (
+        <div className="ocr-error">
+          <p>
+            <Translate>Could not be processed</Translate>
+          </p>
+        </div>
+      );
+      tip = ocrDisplayTips.cantProcess(lastUpdated);
+      break;
+
+    case 'withOCR':
+      statusDisplay = (
+        <div className="status">
+          <p>
+            <Translate>OCR</Translate>&nbsp;&#10004;
+          </p>
+        </div>
+      );
+      tip = ocrDisplayTips.lastUpdated(lastUpdated);
+      break;
+
+    case 'ocrComplete':
+      statusDisplay = (
+        <div className="complete">
+          <p>
+            <Translate>OCR completed</Translate>&nbsp;&#10004;
+          </p>
+        </div>
+      );
+      break;
+
+    default:
+      break;
+  }
+
+  return (
+    ocrIsToggled && (
```

Comment:
The entire component is irrelevant if `orcIsToggled` is off... yet we wait all the logic before "dismissing" the component?  Doesn't it make more sense to put this conditional on line 1 of the component: `if (!ocrIsToggled) return null;` and we prevent all this case switching and stuff if we don't want to render anything?

Reply:
Yes, this totally correct.